### PR TITLE
Fix bug when no pdf found in Public site AB#11711

### PR DESCRIPTION
**Pull request by pagustinqt20**
# Fix bug when no pdf found in Public site [AB#11711](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11711)

-   [ ] Enhancement
-   [x] Bug
-   [ ] Documentation

## Description

Fix bug when no pdf found in Public site

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

Use the below user in Public site : 
9879886741 1987-06-06 LONGFIRSTNAME TESTINGTHELONGESLASTNAMEEEEEEE 1 2021-01-01

**Actual Behavior :**  Always retrying and doesn't display the record. 
**Expected Behavior :** Error should display  and don't retry

### UI Changes

NO

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/getters.ts [hunk 1]
```diff
@@ -17,6 +17,11 @@ export const getters: VaccinationStatusGetters = {
     error(state: VaccinationStatusState): BannerError | undefined {
         return state.public.error;
     },
+    publicVaccineRecordError(
+        state: VaccinationStatusState
+    ): BannerError | undefined {
+        return state.publicVaccineRecord.error;
+    },
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/getters.ts:24]: Nit: might be better to move this after `publicVaccineRecordIsLoading` just for organizational purposes.

File: Apps/WebClient/src/ClientApp/test/stubs/store/vaccinationStatusStub.ts [hunk 2]
```diff
@@ -105,7 +109,7 @@ const vaccinationStatusMutations: VaccinationStatusMutations = {
     setVaccinationStatus: voidMethod,
     vaccinationStatusError: voidMethod,
     setPdfRequested: voidMethod,
-    pdfError: voidMethod,
+    setPublicVaccineRecordError: voidMethod,
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/stubs/store/vaccinationStatusStub.ts:112]: Nit: suggest moving this below with the other `setPublicVaccineRecord*` mutations for organization.

Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/test/stubs/store/vaccinationStatusStub.ts:112]: Also, the `vaccinationStatusActions` above need to be updated to include the new actions.

File: Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/types.ts [hunk 3]
```diff
@@ -132,10 +139,6 @@ export interface VaccinationStatusMutations
         state: VaccinationStatusState,
         statusMessage: string
     ): void;
-    setPublicVaccineRecord(
-        state: VaccinationStatusState,
-        vaccineRecord: CovidVaccineRecord
-    ): void;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/types.ts:138]: I don't think this should be deleted.

File: Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/types.ts [hunk 4]
```diff
@@ -48,6 +49,9 @@ export interface VaccinationStatusGetters
     ): VaccinationStatus | undefined;
     isLoading(state: VaccinationStatusState): boolean;
     error(state: VaccinationStatusState): BannerError | undefined;
+    publicVaccineRecordError(
+        state: VaccinationStatusState
+    ): BannerError | undefined;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/types.ts:54]: Nit: might be better to move this after `publicVaccineRecordIsLoading` just for organizational purposes.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/test/stubs/store/vaccinationStatusStub.ts (hunk 2 above)
Comment (pagustinqt20) [reply Apps/WebClient/src/ClientApp/test/stubs/store/vaccinationStatusStub.ts]: @MikeLyttle  the 1st comment I applied the change but I am bit confused on the 2nd comment?

Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/test/stubs/store/vaccinationStatusStub.ts]: Further up in the file, we mock out the actions for this store module in `const vaccinationStatusActions`, but we forgot to add in the new actions related to the public vaccine record.

File: Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/getters.ts (hunk 1 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/getters.ts:24]: `publicVaccineRecordIsLoading` doesn't need to be moved, just `publicVaccineRecordError`.

**Review by MikeLyttle — APPROVED** [review]
I added a commit that re-orders the actions, mutations, and getters in the vaccinationStatus store interface, implementation, and stub to be consistent.

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**